wicker/core/storage.py

```diff
@@ -72,6 +82,17 @@ def check_exists_s3(self, input_path: str) -> bool:
         except ClientError:
             return False
 
+    @retry(Exception, tries=get_config().s3_storage_config.retries, backoff=5, delay=4, logger=logger)
+    def download_with_retries(self, bucket: str, key: str, local_path: str, s3_input_path: str):
+        try:
+            with time_limit(self.read_timeout):
+                logging.info(f"Trying to download {bucket} {key}")
+                self.client.download_file(bucket, key, local_path)
+                logging.info(f"Downloaded {s3_input_path} on to {local_path}")
+        except Exception as e:
+            logging.error(f"Failed to download s3 path:{s3_input_path}")
+            raise e
+
     def fetch_file_s3(self, input_path: str, local_prefix: str, timeout_seconds: int = 120) -> str:
```

Comment:
Do you want to add an interface for timeout seconds too or are you comfortable with this? Just curious no need to implement

Reply:
I am not sure I get you here. What would an interface would timeout seconds look like?